### PR TITLE
rubocop: Enable RSpec/ContextWording cop & fix offenses

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] ~Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).~
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

- This was TODO, so I thought I'd do it.
- Most interesting is probably the last commit where I disabled the cop for three out of 150 surfaced problems.

### CODE REVIEW

File: Library/Homebrew/test/livecheck/skip_conditions_spec.rb [hunk 1]
```diff
@@ -265,56 +265,56 @@
   end
 
   describe "::skip_conditions" do
-    context "a deprecated formula without a livecheckable" do
+    context "when the formula is deprecated without a livecheckable" do
```
Comment (nandahkrishna) [inline Library/Homebrew/test/livecheck/skip_conditions_spec.rb:268]: I think `"when a formula without a livecheckable is deprecated"` (and similarly for the other tests) conveys the meaning better.

Comment (issyl0) [reply Library/Homebrew/test/livecheck/skip_conditions_spec.rb:268]: Thanks, I updated all of those.

File: Library/Homebrew/test/cask/dsl_spec.rb [hunk 2]
```diff
@@ -168,37 +168,37 @@
         cask.config = config
       end
 
-      context "to 'zh'" do
+      context "when it is 'zh'" do
```
Comment (Rylan12) [inline Library/Homebrew/test/cask/dsl_spec.rb:171]: The wording for these feel a tad off:

```
Cask::DSL language stanza when language is set explicitly when it is 'zh'
```

I almost wonder if the `it` should be allowed for these cases. Either way, it's clear in what it means so not a blocker.

File: Library/Homebrew/test/os/mac/version_spec.rb [hunk 3]
```diff
@@ -40,7 +40,7 @@
     expect(version).to be < Version.create("10.15")
   end
 
-  context "after Big Sur" do
+  context "when after Big Sur" do
```
Comment (Rylan12) [inline Library/Homebrew/test/os/mac/version_spec.rb:43]: Another one that I'm not thrilled with, but is certainly fine to merge.

```
OS::Mac::Version when after Big Sur
```

File: Library/Homebrew/test/requirement_spec.rb [hunk 4]
```diff
@@ -148,7 +148,7 @@
   end
 
   describe "#build?" do
-    context ":build tag is specified" do
+    context "when :build tag is specified" do
```
Comment (Rylan12) [inline Library/Homebrew/test/requirement_spec.rb:151]:
```suggestion
    context "when the :build tag is specified" do
```

Optional

**Review by Rylan12 — APPROVED** [review]
🎉 

I pointed out a few places where I think the wording is sub-optimal (just due to the constraints of the cop rules). I don't think any of them are unclear so I wouldn't bother changing unless anyone has a good idea for how to do it. I couldn't think of any, so I definitely won't complain 😅

File: Library/Homebrew/test/os/mac/version_spec.rb (hunk 3 above)
Comment (issyl0) [reply Library/Homebrew/test/os/mac/version_spec.rb:43]: Yeah I know, some of them are a bit odd.

File: Library/Homebrew/test/cask/dsl_spec.rb (hunk 2 above)
Comment (issyl0) [reply Library/Homebrew/test/cask/dsl_spec.rb:171]: Maybe `Cask::DSL language stanza when language is set explicitly with 'zh'` as a _slightly_ more understandable alternative? Or we can add "to" to the allowed starters (https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ContextWording)?

**Comment by issyl0** [comment]
@Rylan12 Thanks, how does this look?

File: Library/Homebrew/test/cask/dsl_spec.rb [hunk 5]
```diff
@@ -168,37 +168,37 @@
         cask.config = config
       end
 
-      context "when it is 'zh'" do
+      context "to 'zh'" do
```
Comment (Rylan12) [inline Library/Homebrew/test/cask/dsl_spec.rb]:
```suggestion
      describe "to 'zh'" do
```

Sorry, I think I changed my mind on this. Adding `to` to the list of words doesn't really make sense, I think. It doesn't really fit with the rest of the `context` words. I'm basing this partially off of the suggestions [here](https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspeccontextwording). This seems like an exception, so maybe we should just use `describe` here, instead?

Again, though, I'd be fine with merging this as-is. I apologize for changing my mind on this.

Comment (issyl0) [reply Library/Homebrew/test/cask/dsl_spec.rb]: No problem! I also thought about `describe` for these as I was trying to sleep at about 3am. 😂

**Review by Rylan12 — APPROVED** [review]
Looks good to me. Thanks, @issyl0!

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @issyl0!

**Pull request merged**